pkg/component/networking/nodelocaldns/nodelocaldns.go

```diff
@@ -557,3 +566,29 @@ func (n *nodeLocalDNS) SetClusterDNS(dns []string) {
 func (n *nodeLocalDNS) SetDNSServers(servers []string) {
 	n.values.DNSServers = servers
 }
+
+func (n *nodeLocalDNS) SetIPFamilies(ipfamilies []gardencorev1beta1.IPFamily) {
+	n.values.IPFamilies = ipfamilies
+}
+
+func (n *nodeLocalDNS) GetIPVSAddress() (ipvsAddress string) {
+	ipFamiliesSet := sets.New[gardencorev1beta1.IPFamily](n.values.IPFamilies...)
+	if ipFamiliesSet.Has(gardencorev1beta1.IPFamilyIPv4) && !ipFamiliesSet.Has(gardencorev1beta1.IPFamilyIPv6) {
+		ipvsAddress = nodelocaldnsconstants.IPVSAddress
+	}
+	if ipFamiliesSet.Has(gardencorev1beta1.IPFamilyIPv6) && !ipFamiliesSet.Has(gardencorev1beta1.IPFamilyIPv4) {
+		ipvsAddress = nodelocaldnsconstants.IPVSIPv6Address
+	}
+	return
```

Comment:
Please do not use a naked `return` if it returns some value. This can be confusing.
Either use `return ipvsAddress` or do not use a named return value at all, i.e. use a local variable.